src/main/java/org/candlepin/resource/EnvironmentResource.java

```diff
@@ -306,12 +330,13 @@ public ConsumerDTO createConsumerInEnvironment(String envId, ConsumerDTO consume
         this.validator.validateCollectionElementsNotNull(consumer::getInstalledProducts,
             consumer::getGuestIds, consumer::getCapabilities);
 
-        //TODO: This needs to support backward compatibility.
-        // Kept as it is to be handled in other task of multi-environment feature.
-        Environment e = lookupEnvironment(envId);
-        consumer.setEnvironments(Arrays.asList(translator.translate(e, EnvironmentDTO.class)));
+        List<Environment> envs = processEnvironmentIds(envId);
+        consumer.setEnvironments(envs.stream()
```

Comment:
Given that we're calling the process method in one place, and we're using streams to convert everything, we could combine it into one chain:

```
List<EnvironmentDTO> dtos = Arrays.stream(envId.split("\\s*,\\s*"))
    .map(EnvironmentResource::lookupEnvironment)
    .map(this.translator.getStreamMapper(Environment.class, EnvironmentDTO.class))
    .collect(Collectors.toList());

consumer.setEnvironments(dtos);
```

As an aside, I wonder if we shouldn't also verify that all of the environments fetched are in the requested owner. Now that we're just passing the requested owner key through, there's some risk of the owner and environment(s) not lining up if it's never validated.